### ISSUE TITLE
SelectorMethodInjectable return QueryLocator

### ISSUE THREAD

**Issue by alanjaouen** [apex-enterprise-patterns/at4dx#82]
**Is your feature request related to a problem? Please describe.**

When writting apex batch, it's a best practice to use Database.QueryLocator.
I tried to use method injection to inject my own logic in a shared selector like this
```apex
    public Database.QueryLocator start(Database.BatchableContext info) {
        return Application.Selector.newInstance(Account.SObjectType).selectInjection(AccountToProcessMethod.class, null);
    }
```

But `selectInjection` return a `List<SObject>`, as defined in `ISelectorMethodInjectable`.

**Describe the solution you'd like**

I would like be able to inject methods that return `QueryLocator` and not only `List<SObject>`.

**Describe alternatives you've considered**

I considered move my batch from `Database.QueryLocator` to `Iterable<sObject>`, but performance and limits with this are quite boring

**Comment by ImJohnMDaniel** [comment]
G'day @alanjaouen 

Thanks for reaching out with this suggestion.

To be honest, I was thinking about adding this feature myself because I ran into this need on one of my own projects just about a month ago.  The preliminary analysis that I did to determine feasibility suggested that it would not be a heavy lift, which is a good thing.

Let me get your opinion on this.  The method signature that I was thinking about would be the following:
```java
     Database.QueryLocator ql = Application.Selector.newInstance(Account.SObjectType).selectQueryLocatorInjection( {type of ISelectorMethodInjectable}, ISelectorMethodParameterable );
```
Would that be consistent with your expectations of how this would work?  Definitely interested in hearing your thoughts.

Cheers!
